radixdlt/babylon-wallet-ios
# [ABW-3437] Implement deferred deep link solution

Jira ticket: [ABW-3437](https://radixdlt.atlassian.net/browse/ABW-3437)

## Description
This PR resolves deferred deep links via AppsFlyer. This will allow new users to tap on a link that will redirect them to the AppStore, but when the app is downloaded and opened, it will fetch specific information sent on the link query params.
For now, we are just logging this information. In future PRs we will deal with it.

## How to test
Deferred deep links aren't working when tapping the link and then installing the app from TestFlight. Sergio is getting the same behavior on Android, so for now we will run the `pre-alpha` version from Xcode in order to verify it works.

Steps are:
1) Generate and install development certificates if you've never done so.
```
bundle exec fastlane ios generate_new_dev_certificates --env ios.pre-alpha
bundle exec fastlane ios install_development_certificates --env ios.pre-alpha
```

<details>
  <summary> 2. Select development provisioning profile on Xcode</summary>
  
![Screenshot 2024-06-19 at 11 07 39](https://github.com/radixdlt/babylon-wallet-ios/assets/164921079/d7bb1455-861c-444e-9f17-2987e2dcbe34)
</details>

3) Replace values on `SensitiveInfo.plist` so that AppsFlyer uses the `appId` & `devKey` for pre-alpha project. You should be able to see those in [AppsFlyer console](https://hq1.appsflyer.com/apps/settings). Ping me on Slack if you need help with this.

4) Delete every Radix Wallet app from your device (should work just by deleting pre-alpha, but I've tested without any other)

5) Select any One Link on the [AppsFlyer dashboard](https://hq1.appsflyer.com/onelink) from `test` template. For example, you can select:
```
https://radixdlt.onelink.me/WAKF/eun9mos3
```

6) Append the query parameters you would like to catch under `deep_link_value` or `deep_lin_sub1/10`. For example, it should like as:
```
https://radixdlt.onelink.me/WAKF/eun9mos3?deep_link_value=reviewPR&deep_link_sub1=thisWorks
```

7) Copy the link on your device (e.g. in Notes app), and tap on it. Your browser will open indicating `The app you are looking for is unavailable`. This is because pre-alpha isn't available on AppStore.

8) Run the app from Xcode to simulate an install.

9) Check the logs and filter by `click events`. You should see something like the following:
```
2024-06-19T17:32:11+0200 🟦 info com.radixpublishing : did resolve deep link. Is deferred: true. 
Click events: ["match_type": probabilistic, "af_sub4": , "click_http_referrer": , "af_sub1": , "af_sub3": , 
"campaign": , "deep_link_value": reviewPR, "af_sub2": , "campaign_id": , "media_source": , "af_sub5": , 
"deep_link_sub1": thisWorks]
```

You can see that both query paramters have been intercepted by the app after its installation.


[ABW-3437]: https://radixdlt.atlassian.net/browse/ABW-3437?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
